### DISCUSSION TITLE
Redeployment failed

### DISCUSSION THREAD

**Discussion by positivism1** [Azure/ipam#158] [General]
Got a strange issue where I've redeployed IPAM and now I can't access via browser or API.
API I get Bad Gateway 502
Browser I get the spinning circle "loading page"
No obvious errors in the logs.

**Comment by positivism1** [comment]
Found an error in the Availability and Performance Blade
Your Docker image may not exist or your container registry settings are not correct.
Image pull failed since inspect image returned null  <name of acr>.azurecr.io/ipam-ui:latest

I'm going to try re-download the repo and install again. 
I'm using Private ACR

**Comment by DCMattyG** [comment]
Hi @positivism1, I'm glad you located the issue. I'd be interested in how you deployed the IPAM solution (what PowerShell switches did you use). The fact that the result was a blank name where the Private ACR is supposed to be in Docker Compose is concerning and if it's a bug in the script, I'd like to get it addressed right away.

Please let me know if you encounter any issues once you deploy the solution again, and I'll try to re-create the issue on my end (if possible).

Thank you for reaching out, we're always here to help!

**Comment by positivism1** [comment]
It's a strange issue. My thinking is that the application changed two weeks ago and i've got an older copy that i'm using to deploy it. 
I've updated to the latest deploy folder now getting this new error I haven't seen before.

![image](https://github.com/Azure/ipam/assets/3076159/6af5772f-2753-4961-a810-5cfe40f1a4de)


Failing forward.. 
![image](https://github.com/Azure/ipam/assets/3076159/415fbb24-ceef-429f-b47a-41fad8883edb)

**Reply by DCMattyG** [reply]
Ok, I know exactly what the issue is and I will fix this right away...

The problem is that I added "pip install --upgrade pip", but this will try to display a "progress bar" in the console. This causes the error that you're seeing above. I need to add the "--progress-bar off" switch to suppress that behavior.

You can validate this by adding the "--progress-bar off" switch to the following file and attempting the deployment again:
- engine\Dockerfile.deb -> Line 16

Please let me know if that fixes the issue. I'll get a fix into the repo ASAP.

**Comment by positivism1** [comment]
yay Pammy is back! 
![image](https://github.com/Azure/ipam/assets/3076159/e7f60e86-ad10-405a-bce1-c8fbd63cffae)

What an ordeal, let's agree to never change anything again. 🤣
Thanks @DCMattyG !!

**Reply by DCMattyG** [reply]
Hahaha! The really good news is that my very next update is GitHub Actions that deploy IPAM and perform end-to-end testing every time a change is pushed to main so we can avoid this exact scenario in the future 😁
